### ISSUE TITLE
Yet another cursor bug

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#86]
### Description

If you move cursor in source code mode we got "random" characters in marked.js `src`. Works with commit `216c522` but I didn't really look into it yet. 

### Steps to reproduce

1. Set `console.log(src);` [here](https://github.com/marktext/marktext/blob/60ac2b675bbf409ddc3bf44823dcb3f2c29fa4cd/src/editor/parser/marked.js#L161)
2. Write some text
3. Move cursor position

**Expected behavior:**

No garbage in marked.js parser.

**Actual behavior:**

![cursor-bug](https://user-images.githubusercontent.com/22716132/37999275-678e8cda-3222-11e8-9c2e-69a865b4e308.gif)

### Versions

- Mark Text: ? (past `216c522`)
- OS: linux

**Comment by Jocs** [comment]
@fxha I use the random characters to record the cursor position when translate cursor in source code mode to cursor in contentState which we use to render the content state.

Do you have any better ideas to record the cursor position?

**Comment by fxha** [comment]
Maybe use a variable or something because following list will "fail" to parse:

```
- a
<cursor>
- b
```

It's a loose list, but marked.js just see a tight list.

```
- a
%random%
- b
```

**Comment by Jocs** [comment]
@fxha yeah, maybe need adjust the cursor again.
when cursor is:

```
- a
<cursor>
- b
```

change to:
```
- a<cursor>

- b
```

What's your opinions?

**Comment by fxha** [comment]
Well, that was just an example. I don't know if this helps or not. Same here:

```
\```
foobar();
\```<cursor>

becomes

\```
foobar();ag-8lfiqghseqo-1522245547053
\```
```

So far so good, but:

```
\```
foobar();

\```<cursor>

becomes

\```
foobar();
\```ag-8lfiqghseqo-1522245547053
```
... and fails to parse.

**Comment by Jocs** [comment]
@fxha I will check and fix this bug after #74 merged.
